cranelift/codegen/src/machinst/mod.rs

```diff
@@ -164,14 +167,31 @@ pub trait MachInst: Clone + Debug {
     /// Generate a move.
     fn gen_move(to_reg: Writable<Reg>, from_reg: Reg, ty: Type) -> Self;
 
-    /// Generate a constant into a reg.
+    /// Determine whether a constant value should be generated in place. Higher-level users can use
+    /// this to determine whether `gen_constant` or `gen_constant_in_pool` should be used.
+    fn should_generate_constant_in_pool(_value: &DataValue) -> bool {
+        false
+    }
+
+    /// Generate a constant into a register. `alloc_tmp_reg` must be available because some
+    /// instruction sequences generating a constant value require an extra register (e.g. for moving
+    /// bits from general purpose to floating point registers)
     fn gen_constant<F: FnMut(RegClass, Type) -> Writable<Reg>>(
```

Comment:
Perhaps we can call this `gen_constant_in_place` now (analogous to ...`_in_pool` below)?